src/models/tree-api.ts

```diff
@@ -0,0 +1,96 @@
+import { IJsonPatch } from "mobx-state-tree";
+
+/**
+ * This is the API for a Tree in the system.
+ *
+ * It would typically be implemented by a MST model that defines actions for
+ * each fo the functions below.
+ *
+ * Each action should return a promise that resolves when the action is complete
+ * this is necessary to support tress running in iframes or workers. The
+ * function comment should define what "complete" means for each action.
+ */
+
+export interface TreeAPI {
+    /**
+     * This is called when the container is doing an undo or redo. The tree
+     * should use this action to disable any updating it does when it receives
+     * changes in the shared models it is using.
+     *
+     * @returns a promise that should resolve when the tree is ready to receive
+     * patches from the container and changes in the shared models.
+     *
+     * The `Tree` model implements for you.
```

Comment:
I'm not sure how to interpret this sentence.
```suggestion
     * The `Tree` model implements this for you.
```
Down below there's the same comment with the additional "this".